src/disko/mode_generate.py

```diff
@@ -0,0 +1,75 @@
+import json
+import re
+from disko_lib.eval_config import eval_config_dict_as_json
+from disko_lib.logging import info
+from disko_lib.result import DiskoPartialSuccess, DiskoResult, DiskoError
+from disko_lib.generate_config import generate_config
+from disko_lib.run_cmd import run
+from disko_lib.json_types import JsonDict
+
+DEFAULT_CONFIG_FILE = "disko-config.nix"
+
+HEADER_COMMENT = """
+# This file was generated by disko generate
+# Some disk and partition names were auto-generated from device attributes
+# to be unique, but might not be very descriptive. Feel free to change them
+# to something more meaningful.
+
+"""
+
+PARTIAL_FAILURE_COMMENT = """
```

Comment:
Add a backslash if you don't want an empty first line.

```suggestion
PARTIAL_FAILURE_COMMENT = """\
```

Reply:
Ah right, totally forgot about this.